financial-connections/src/main/java/com/stripe/android/financialconnections/image/StripeImage.kt

```diff
@@ -0,0 +1,169 @@
+    url: String,
+    placeHolder: Painter,
+    memoryCache: LruCache<String, Bitmap> = imageMemoryCache,
+    contentDescription: String? = null,
+    modifier: Modifier = Modifier
+) {
+    BoxWithConstraints(modifier) {
+        val (width, height) = calculateBoxSize()
+        val painter: MutableState<Painter> =
+            remember { mutableStateOf(BitmapPainter(ImageBitmap(width, height))) }
+        var loadImageJob: Job?
+        val scope = rememberCoroutineScope()
+
+        DisposableEffect(url) {
+            val imageLoader = ImageLoader()
+            loadImageJob = scope.launch(Dispatchers.Default) {
+                painter.value = memoryCache.get(url)
+                    ?.let { BitmapPainter(it.asImageBitmap()) }
+                    ?: imageLoader.load(url, width, height)
+                        .fold(
+                            onSuccess = { bitmap ->
+                                debug("Image loaded")
+                                memoryCache.put(url, bitmap)
+                                BitmapPainter(bitmap.asImageBitmap())
+                            },
+                            onFailure = {
+                                debug("Image failed loading")
+                                it.printStackTrace()
+                                placeHolder
+                            }
+                        )
+
+            }
+            onDispose {
+                painter.value = BitmapPainter(ImageBitmap(width, height))
+                imageLoader.cancel()
+                loadImageJob?.cancel()
+            }
+        }
+        Image(
+            modifier = modifier,
+            contentDescription = contentDescription,
+            painter = painter.value
+        )
+    }
+}
+
+fun debug(message: String) {
+    Log.d("StripeImage", message)
+}
+
+private fun BoxWithConstraintsScope.calculateBoxSize(): Pair<Int, Int> {
+    var width =
+        if (constraints.maxWidth > Zero.width &&
+            constraints.maxWidth < Infinity.value.toInt()
+        ) {
+            constraints.maxWidth
+        } else {
+            -1
+        }
+
+    var height =
+        if (constraints.maxHeight > Zero.height &&
+            constraints.maxHeight < Infinity.value.toInt()
+        ) {
+            constraints.maxHeight
+        } else {
+            -1
+        }
+
+    // if height xor width not able to be determined, make image a square of the determined dimension
+    if (width == -1) width = height
+    if (height == -1) height = width
+    return Pair(width, height)
+}
+
+// TODO@carlosmuvi scope this to SDK lifecycle and remove static reference.
+private val imageMemoryCache = object : LruCache<String, Bitmap>(
+    // Use 1/8th of the available memory for this memory cache.
+    (Runtime.getRuntime().maxMemory() / 1024).toInt() / 8
+) {
+    override fun sizeOf(key: String, bitmap: Bitmap): Int {
+        return bitmap.byteCount / 1024
+    }
+}
+
+private class ImageLoader {
```

Comment:
WHat do you think about moving the ImageLoader nearer to the other API stuff so that we can inject an analytics executor.  This would allow us to post events if the image isn't found.